txp/Pos/Txp/Toil/Failure.hs

```diff
@@ -32,43 +32,27 @@ import           Pos.Util (DisallowException)
 -- | Result of transaction processing
 data ToilVerFailure
     = ToilKnown -- ^ Transaction is already in the storage (cache)
-    | ToilTipsMismatch { ttmOldTip :: !HeaderHash
-                       , ttmNewTip :: !HeaderHash}
+    | ToilTipsMismatch !HeaderHash !HeaderHash
     | ToilSlotUnknown
     | ToilOverwhelmed !Int -- ^ Local transaction storage is full --
                             -- can't accept more txs. Current limit is attached.
     | ToilNotUnspent !TxIn -- ^ Tx input is not a known unspent input.
-    | ToilOutGreaterThanIn { tInputSum  :: !Integer
-                           , tOutputSum :: !Integer}
+    | ToilOutGreaterThanIn !Integer !Integer
     | ToilInconsistentTxAux !Text
     | ToilInvalidOutput !Word32 !TxOutVerFailure
     | ToilUnknownInput !Word32 !TxIn
-
     -- | The witness can't be used to justify spending an output – either
     --     * it has a wrong type, e.g. PKWitness for a script address, or
     --     * it has the right type but doesn't match the address, e.g. the
     --       hash of key in PKWitness is not equal to the address.
-    | ToilWitnessDoesntMatch { twdmInputIndex  :: !Word32
-                             , twdmInput       :: !TxIn
-                             , twdmSpentOutput :: !TxOut
-                             , twdmWitness     :: !TxInWitness }
-
+    | ToilWitnessDoesntMatch !Word32 !TxIn !TxOut !TxInWitness
     -- | The witness could in theory justify spending an output, but it
     -- simply isn't valid (the signature doesn't pass validation, the
     -- validator–redeemer pair produces 'False' when executed, etc).
-    | ToilInvalidWitness { tiwInputIndex :: !Word32
-                         , tiwWitness    :: !TxInWitness
-                         , tiwReason     :: !WitnessVerFailure }
-
-    | ToilTooLargeTx { ttltSize  :: !Byte
-                     , ttltLimit :: !Byte}
-    | ToilInvalidMinFee { timfPolicy :: !TxFeePolicy
-                        , timfReason :: !Text
-                        , timfSize   :: !Byte }
-    | ToilInsufficientFee { tifPolicy :: !TxFeePolicy
-                          , tifFee    :: !TxFee
-                          , tifMinFee :: !TxFee
-                          , tifSize   :: !Byte }
+    | ToilInvalidWitness !Word32 !TxInWitness !WitnessVerFailure
+    | ToilTooLargeTx !Byte !Byte
+    | ToilInvalidMinFee !TxFeePolicy !Text !Byte
+    | ToilInsufficientFee !TxFeePolicy !TxFee !TxFee !Byte
```

Comment:
When I see these names: 

```
    | ToilInsufficientFee { tifPolicy :: !TxFeePolicy
                          , tifFee    :: !TxFee
                          , tifMinFee :: !TxFee
                          , tifSize   :: !Byte }
```

I can understand, for example, that second field is a `fee` and third one is a `minFee`. But when I see this:

```
| ToilInsufficientFee !TxFeePolicy !TxFee !TxFee !Byte
```

I don't see any differences between second field and third one. :-)

So, probably just use type aliaces, no? For example:

```
type TxMinFee = TxFee
type Size = Byte
...

| ToilInsufficientFee !TxFeePolicy !TxFee !TxMinFee !Size
```

In this case I immediately understand what's going on. Do you agree?

Reply:
> I added comments which provided the same level of type safety as type aliases :)

:-)